external_server/adapters/api/module_lib.py

```
@@ -0,0 +1,163 @@
+
+    def get_module_number(self) -> int:
+        with self._lock:
+            return int(self.library.get_module_number())
+
+    def init(self) -> ct.c_void_p:
+        if not os.path.isfile(self._lib_path):
+            raise FileNotFoundError(self._lib_path)
+        try:
+            self._library = ct.cdll.LoadLibrary(self._lib_path)  # type: ignore
+        except OSError as e:
+            raise RuntimeError(f"Failed to load library '{self._lib_path}': {e}")
```

Ruff

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)